### PR TITLE
Better mod support

### LINKED ISSUE #118: Parsing issue
`Exception thrown: 'System.IndexOutOfRangeException' in Parser.dll
2021-05-17 09:47:57,261 [5] WARN  IAGrim.Parser.Arc.DDSImageReader Error extracting icon "chests/anm/chestsandtemple_01a_idle_a01.anm", Index was outside the bounds of the array.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at EvilsoftCommons.IOHelper.GetInt(Byte[] data, Int32 pos) in F:\Dev\IAGrim\Parser\IOHelper.cs:line 153
   at IAGrim.Parser.Arc.DDSImageReader.getDDSHeader(Byte[] bytes) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 187
   at IAGrim.Parser.Arc.DDSImageReader.ExtractImage(Byte[] bytes) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 94
   at IAGrim.Parser.Arc.DDSImageReader.ExtractItemIcons(String itemsArcFullpath, String destinationFolder) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 67
Exception thrown: 'System.IndexOutOfRangeException' in Parser.dll
2021-05-17 09:47:57,272 [5] WARN  IAGrim.Parser.Arc.DDSImageReader Error extracting icon "chests/anm/sarcophagus_sandtemple01a_idle_a01.anm", Index was outside the bounds of the array.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at EvilsoftCommons.IOHelper.GetInt(Byte[] data, Int32 pos) in F:\Dev\IAGrim\Parser\IOHelper.cs:line 153
   at IAGrim.Parser.Arc.DDSImageReader.getDDSHeader(Byte[] bytes) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 187
   at IAGrim.Parser.Arc.DDSImageReader.ExtractImage(Byte[] bytes) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 94
   at IAGrim.Parser.Arc.DDSImageReader.ExtractItemIcons(String itemsArcFullpath, String destinationFolder) in F:\Dev\IAGrim\Parser\Arc\DDSImageReader.cs:line 67`

### PR DESCRIPTION
**Pull request by Eradev**
Better mod support, at least for Dawn of Masteries.

Also fixes #118

### CODE REVIEW

File: IAGrim/Database/DAO/ItemTagDaoImpl.cs [hunk 1]
```diff
@@ -37,35 +39,47 @@ class ItemTagDaoImpl : BaseDao<ItemTag>, IItemTagDao {
...
         }
 
         public ISet<ItemTag> GetValidClassItemTags() {
+            const string gdxClassesPattern = @"tagGDX\dClass(\d+)SkillName00A";
+            const string namePattern = @"\[ms\](.*)\[fs\](.*)";
+
             using (var session = SessionCreator.OpenStatelessSession()) {
                 using (session.BeginTransaction()) {
-                    return session.CreateSQLQuery("SELECT * FROM ItemTag WHERE (Tag LIKE 'tagSkillClassName%' OR Tag LIKE 'tag%Class%SkillName00A') AND LENGTH(Name) > 1")
+                    return session
```
Comment (marius00) [inline IAGrim/Database/DAO/ItemTagDaoImpl.cs:67]: IDE not respecting the line length in .editorconfig?

Not the end of the world, but a bit odd.

File: IAGrim/UI/Tabs/ModsDatabaseConfig.cs [hunk 2]
```diff
@@ -140,6 +141,12 @@ public void ForceDatabaseUpdate(string location, string modLocation)
                 var mod = listViewMods.SelectedItems[0].Tag as ListViewEntry;
                 var entry = lvi.Tag as ListViewEntry;
 
+                if (mod != null)
+                {
```
Comment (marius00) [inline IAGrim/UI/Tabs/ModsDatabaseConfig.cs:145]: Really looks like it ignored editorconfig on this one, but no biggie.

File: IAGrim/UI/Tabs/ModsDatabaseConfig.cs [hunk 3]
```diff
@@ -140,6 +141,12 @@ public void ForceDatabaseUpdate(string location, string modLocation)
...
+                    // Load selected mod icons
+                    ThreadPool.QueueUserWorkItem((m) => ArzParser.LoadSelectedModIcons(mod.Path));
```
Comment (marius00) [inline IAGrim/UI/Tabs/ModsDatabaseConfig.cs:147]: Doing it parallel to the database parse will most likely cause an OOM on lower end PCs.

File: Parser/Arz/ArzParser.cs [hunk 4]
```diff
@@ -38,20 +38,35 @@ public class ArzParser {
 
         private static bool IsInteresting(string record) {
             string[] interesting = {
-                "/scriptentities/",
+                "/scriptentities/", // TODO: Why?
```
Comment (marius00) [inline Parser/Arz/ArzParser.cs:41]: Totally Normal Shield

File: IAGrim/UI/Tabs/ModsDatabaseConfig.cs (hunk 3 above)
Comment (Eradev) [reply IAGrim/UI/Tabs/ModsDatabaseConfig.cs:147]: Any suggestions to an alternative?

File: Parser/Arz/ArzParser.cs (hunk 4 above)
Comment (Eradev) [reply Parser/Arz/ArzParser.cs:41]: Gotcha.

File: IAGrim/Database/DAO/ItemTagDaoImpl.cs (hunk 1 above)
Comment (Eradev) [reply IAGrim/Database/DAO/ItemTagDaoImpl.cs:67]: You meant line 69? Can probably cut it if you want.

File: IAGrim/UI/Tabs/ModsDatabaseConfig.cs (hunk 2 above)
Comment (Eradev) [reply IAGrim/UI/Tabs/ModsDatabaseConfig.cs:145]: Fixed.

File: IAGrim/Database/DAO/ItemTagDaoImpl.cs (hunk 1 above)
Comment (marius00) [reply IAGrim/Database/DAO/ItemTagDaoImpl.cs:67]: Not an issue, just a curiosity, that it didnt respect the editorconfig.

File: IAGrim/UI/Tabs/ModsDatabaseConfig.cs (hunk 3 above)
Comment (marius00) [reply IAGrim/UI/Tabs/ModsDatabaseConfig.cs:134]: As per discord: Will see how it turns out.

**Pull request merged**